Remove guice binding for Linguistics.

gjoranv opened the pull request:
- The config model adds a SimpleLinguisticsProvider for all
  container clusters.

gjoranv committed:
Remove guice binding for Linguistics.

- The config model adds a SimpleLinguisticsProvider for all
  container clusters.

baldersheim commented:
The PR build fails some related tests. Does it mean it can not be fixed, or do we just need to update the tests.

gjoranv commented:
I'm looking into the tests. I think it can/should be fixed there.

gjoranv committed:
Add binding for Linguistics to HandlersConfigurerTestWrapper.

- The binding has been removed from HandlersConfigurerDi to
  prevent creating a SimpleLinguistics there.

gjoranv committed:
Create SimpleLinguistics singleton lazily.

gjoranv commented:
We can try to merge if the build is green. If it fails somewhere, we can use the method from the last commit in `HandlersConfigurerDi` instead.
